clickanalysis/middleware/__init__.py

```diff
@@ -10,32 +13,22 @@ class ClickAnalysisMiddleware(object):
     """
 
     def process_request(self, request):
+
         if request.GET.get('cid', ''):  # Will process the logic only if campaign id is present
             try:
                 campaign = Campaign.objects.get(campaign_id=request.GET['cid'])
-            except:
+            except Campaign.DoesNotExist:
                 campaign = None
 
             # Clearing the cid parameter from the link or providing a clean url for user to redirect
-            get_params = request.get_full_path().split('?')[1].split('&')
-            params_to_send = []
-            for g_p in get_params:
-                name = g_p.split('=')[0]
-                if name != 'cid':
-                    params_to_send.append(g_p)
-
-            params = '&'.join(params_to_send)
-            if params_to_send:
-                url = request.get_full_path().split('?')[0] + '?' + params
-            else:
-                url = request.get_full_path().split('?')[0]
-
-            if campaign:  # If campaign was found then here the click will be recorded
+            url_parsed = urlparse(request.get_full_path())
+            get_params = parse_qsl(url_parsed.query)
+            params_to_send = urlencode([(k, v) for k, v in get_params if k != 'cid'])
+            url = urlunsplit(url_parsed[:3] + (params_to_send,) + url_parsed[5:])
+
+            if campaign is not None:  # If campaign was found then here the click will be recorded
```

Comment:
Ah!, I thought you were asking about the code snippet [GitHub is showing me](https://db.tt/iq31F9Xa).

I am not taking about my personal programming practice in Python, everyone follows PEP-008 style guide unless your company has a style-guide that recommends using `if obj` over `if obj is not None` then you can stick with it. Any seasoned Python programmer will recommend against it. `None` and `object()` when used as sentinel value should always be checked using `is` not a simple `if` which can fail for any falsy value(even `==` shouldn't be used, as a singleton we can directly check its identity rather than calling its `__eq__`, in CPython at most places usually [identity is checked before value comparison](https://docs.python.org/2/c-api/object.html#c.PyObject_RichCompareBool)).

**Example 1:**

    def func(a=None):
        if a:  # Now this fill fail for [], '', {}, etc but if we had used `a is not None` then it would've work fine.
            print 'foo'

**Example 2:**

```
>>> def foo():        
    return ''
... 
>>> out = foo()        
>>> if out:            
    print 'foo returned {!r}'.format(out)
else:
    print 'foo returned nothing'
...     
foo returned nothing        # Eh but it actually did!!
>>> if out is not None:
    print 'foo returned {!r}'.format(out)
else:
    print 'foo returned nothing'
...     
foo returned ''
```

Using plain conditions can lead to such issues.